internal/services/mobilenetwork/mobile_network_service_resource.go

```diff
@@ -0,0 +1,741 @@
+package mobilenetwork
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/mobilenetwork/2022-11-01/mobilenetwork"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/mobilenetwork/2022-11-01/service"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type ServiceModel struct {
+	Name                         string                      `tfschema:"name"`
+	MobileNetworkMobileNetworkId string                      `tfschema:"mobile_network_id"`
+	Location                     string                      `tfschema:"location"`
+	PccRules                     []PccRuleConfigurationModel `tfschema:"pcc_rule"`
+	ServicePrecedence            int64                       `tfschema:"service_precedence"`
+	ServiceQosPolicy             []QosPolicyModel            `tfschema:"service_qos_policy"`
+	Tags                         map[string]string           `tfschema:"tags"`
+}
+
+type PccRuleConfigurationModel struct {
+	RuleName                 string                         `tfschema:"name"`
+	RulePrecedence           int64                          `tfschema:"precedence"`
+	RuleQosPolicy            []PccRuleQosPolicyModel        `tfschema:"qos_policy"`
+	ServiceDataFlowTemplates []ServiceDataFlowTemplateModel `tfschema:"service_data_flow_template"`
+	TrafficControlEnabled    bool                           `tfschema:"traffic_control_enabled"`
+}
+
+type PccRuleQosPolicyModel struct {
+	AllocationAndRetentionPriorityLevel int64             `tfschema:"allocation_and_retention_priority_level"`
+	QosIdentifier                       int64             `tfschema:"qos_indicator"`
+	GuaranteedBitRate                   []maxBitRateModel `tfschema:"guaranteed_bit_rate"`
```

Comment:
Should this be `[]guaranteedBitRateModel`?

Reply:
renamed to `BitRateModel` as they are same Model indeed